src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
```

Comment:
I think that there are cases where an `enum` makes sense. As @scottschurr mentioned, having three or four `bool` variables in a function parameter list is a problem but at the same time I'm not sure I want a proliferation of aliases to `bool`.

Our first instinct should be: why is a `bool` argument needed here? Something smells and can the design be changed to eliminate the argument?